### ISSUE TITLE
No way to all-caps just the beginning of a word

### ISSUE THREAD

**Issue by arxanas** [openstenoproject/plover#837]
Classification: Bug

Reproducibility: Always

## Summary

Suppose you have the dictionary

```
"PEUD": "pid",
"TPAOEUL": "file",
"KPA*TS": "{*<}",
"H*PB": "{^-^}",
"TK-LS": "{^^}",
```

and you try to write PID-file:

```
PEUD/KPA*TS/H*PB/TK-LS/TPAOEUL
```

This actually produces

```
PID-FILE
```

Instead of

```
PID-file
```

In general there doesn't seem to be a way to make capitalizing the previous part of the word not affect the next part of the word.

## Version

Plover version 4.0.0.dev5 @ ff51971a25c570dcc80146d6062cc4e28ed11864
Installed via: source

## Configuration

```
ProductName:    Mac OS X
ProductVersion: 10.12.6
BuildVersion:   16G29
```

**Comment by benoit-pierre** [comment]
This is an intentional change from previous versions, the question is do we want something like this to work:
```python
    def test_retro_upper7(self):
        r'''
        'TEFT': 'test',
        "W-G": "{^ing with}",
        'RUP': '{*<}',

        TEFT/RUP/W-G  " TESTING with"
        '''
```
?

**Comment by arxanas** [comment]
@benoit-pierre: What's the recommended way of writing a word like "PID-file", then? I generally use retroactive capitalization instead of preemptive capitalization, and it doesn't seem like there's a good way to accomplish this anymore, since I can't even use `TK-LS` to cancel the capitalization.

**Comment by morinted** [comment]
I think the example in the OP is more realistic than the "test^ing with" issue. I'd probably still expect 
"TESTing with" due to the location of RUP.

**Comment by arxanas** [comment]
This also appears to be manifesting when I'm coding. I tried to write this

```
  return SomeEnum::SOME_VALUE;
}

public
```

but got

```
  return SomeEnum::SOME_VALUE;
}

PUBLIC
```

where I typed

```
SOM/RUND/VAOUL/KPA*TS # SOME_VALUE
STPH*FPLT  # ;
STPH*B/STPH*B  # down-arrow a few times
R*R/R*R
PLIK
```

where `R*R` is defined to be `{MODE:RESET}{^}\n{^}`. There's even an explicit mode-reset in there. I also tried issuing explicit mode-resets with my `KLAO*ER` stroke.

What exactly am I supposed to do at this point? Go back up to the previous point in the code and delete everything I just typed and then try again, this time without using capitalize-last-word? Give up and restart Plover? It's not even as if I can reset the current mode and continue at the current position.

**Comment by morinted** [comment]
Fixed by #865

**Comment by gdwaynewarner** [comment]
My Phoenix Theory Medical dictionary has P*-EUD for "PID." So if I wanted "PID-file," I would just stroke that first outline, the hyphen stroke (P-F), and, of course, follow it up with my stroke for "file."

Looks like you're working way too hard for an outline that is actually in steno-order! :o)
